src/meili/map_matcher.cc

```diff
@@ -637,6 +640,21 @@ MapMatcher::OfflineMatch(const std::vector<Measurement>& measurements, uint32_t
       std::cout << R"(]})" << std::endl;*/
       best_paths.emplace_back(std::move(match_results));
     }
+
+    // If we just got some results then we need to remove them from consideration
+    // we avoid doing this for the common case where k==1 by putting this at the start of the loop
+    if (!found_broken_path) {
```

Comment:
this is unneccessary if `k==1` which is the vast majority of cases. id really like to avoid calling any of this in that case, that is why i had it at the top of the loop only if we've already gotten more than one result

Reply:
Fixed.